atoms/_forms.scss

```diff
@@ -120,11 +127,16 @@ $input-disabled-background-color: $catkin !default;
       right: 0;
       top: 50%;
       margin-top: -8px;
+      pointer-events: none;
 
       &.icon24 {
         margin-top: -12px;
       }
     }
+
+    input.is-invalid + .icon {
```

Comment:
using `+` selector looks hacky to me. Could lead to some unexpected behavior. 
Can we somehow avoid it?

Reply:
Why does it look hacky to you? It's a common selector and it just affects the icon next to the input.

Reply:
Because selector is referring to any icon which is placed next to `input`, but not every icon which placed next to input should be painted red when input is invalid. For example if it is question mark icon next to the input to clarify what is actually input value stands for.

Also not in all cases icon which goes after the input is actually referring to this input.
It's probably an edge case but still let's take it as an example:
```html
<input name='somedata' style='float: left'/>
<span class='icon icon-dropdown'  style='float: right'> some another dropdown icon to show not related data</span>
```
In this case both of this UI elements will be located on different sides of the screen and will play different roles.
And in the end this is a selector which can not be autoprefixed which leads to a problems I already described before.

About better ways:
If we really want our UI system to support inputes combined with icons then we should probably introduce a wrapper for them like [bootstrap](https://getbootstrap.com/docs/4.0/components/input-group/) and [foundation](https://foundation.zurb.com/sites/docs/forms.html#inline-labels-and-buttons) do.
If combining icon with input is not such common case it's maybe doesn't worth to add such feature to UI library but instead implement it inside app which needs it.

Reply:
Ehm there is a wrapper class called .input-wrapper. Only when this class is grouping an icon and an input this rule applies. I think this is a pretty good pattern to on the one side allow freedom to the developers but on the other side give a guideline how to display things.

Reply:
@fragsalat sorry my bad, you are right, haven't noticed this :snail: 
Then it's not such a big deal but still the thing I would like to see in fabric css is when I'm putting class on some tag then it will change this tag style but not style of it's neighbor.
It's like explicit over implicit. In this case I would expect `.is-invalid` class on wrapper not on input itself. 
```css
.input-wrapper.is-invalid .icon {
   color: red;
}
```
Looks much more explicit for me

But as I said not such a big deal now... Approving this :+1:

Reply:
Fun fact I already had it like this but changed it for any reason. The good thing is that nothing is fixed and everything can be refactored :)